modules/lang-painless/src/main/java/org/elasticsearch/painless/node/PSubDefArray.java

```diff
@@ -59,13 +59,10 @@ void analyze(Locals locals) {
 
     @Override
     void write(MethodWriter writer, Globals globals) {
-        index.write(writer, globals);
```

Comment:
Also, I realize this was probably copied from PSubBrace which has the ```if (!write)``` guard, but it's actually not necessary.  I think it got left over from PSubBrace accidentally when I was working on the EAssignment refactor, so it can be removed there too.

You can see in setup method for these classes that the prefix node is appropriately called while skipping the write method altogether for assignments on the final lhs node.